doc/VTTabletModes.md

```diff
@@ -0,0 +1,101 @@
+VTTablet can be configured to control mysql at many levels. At the level with the most control, vttablet can perform backups and restores, respond to reparenting commands coming through vtctld, automatically fix replication, and enforce semi-sync settings.
+
+At the level with the least control, vttablet just sends the application’s queries to mysql. The level of desired control is achieved through various command line arguments, explained below.
+
+## Managed MySQL
+
+In the mode with the highest control, VTTablet can take backups. It can also automatically restore from an existing backup to prime a new replica. For this mode, vttablet needs to run on the same host as mysql, and must be given access to mysql's my.cnf file. Additionally, the flags must not contain any connectivity parameters like `-db_host` or `-db_socket`; VTTablet will fetch the socket information from my.cnf and use that to connect to the local mysql.
+
+It will also load other information from the my.cnf, like the location of data files, etc. When it receives a request to take a backup, it will shut down mysql, copy the mysql data files to the backup store, and restart mysql.
+
+The my.cnf file can be specified the following ways:
+
+* Implicit: If mysql was initialized by the `mysqlctl` tool, then vttablet can find it based on just the `-tablet-path`. The default location for this file is `$VTDATAROOT/vt_<tablet-path>/my.cnf`.
+* `-mycnf-file`: This option can be used if the file is not present in the default location.
+* `-my_cnf_server_id` and other flags: You can specify all values of the my.cnf file from the command line, and vttablet will behave as it read this information from a physical file.
```

Comment:
You should specify the three files that form the base my.cnf (master.cnf,default.cnf,replica.cnf). Also aside from `mycnf-file` you can pass in the  EXTRA_MY_CNF env flag to add in additional files. And at some place, probbably not here we should list out how vitess uses the go template language to construct the final my.cnf

Reply:
These are for mysqlctl. VTTablet doesn't care about them. I'll remember to mention them while documenting mysqlctl.